### PR TITLE
Make not being able to set target permissions a warning.

### PR DESCRIPTION
**Pull request by mefyl**
Fix issue #3714 .

We could warn once per filesystem instead of once and for all, but that sounds like a lot of work esp portability wise for a very small improvement.
We should provide the user with a way to disable removing write permission so they don't see this warning ad vitam eternam if their filesystem don't allow it. This could be a `(remove_target_write_permissions enabled/disabled)` stanza in the dune configuration file.

### CODE REVIEW

File: src/dune/cached_digest.ml [hunk 1]
```diff
@@ -103,14 +103,23 @@ let refresh fn =
...
   let stats = Path.stat fn in
   let () =
     (* We remove write permissions to uniformize behavior regardless of whether
        the cache is activated. No need to be zealous in case the file is not
        cached anyway. See issue #3311. *)
     if Cache.cachable stats.st_kind then
-      Path.chmod ~stats:(Some stats) ~mode:0o222 ~op:`Remove fn
+      try
+        Path.chmod ~stats:(Some stats) ~mode:0o222 ~op:`Remove fn
+      with _ ->
```
Comment (dra27) [inline src/dune/cached_digest.ml:117]: This is one of those instances where you can catch `Sys.Break` when you didn't mean to. I think catching `Unix_error` is sufficient.

Comment (mefyl) [reply src/dune/cached_digest.ml:117]: Ah, didn't know about `Sys.break`. Fixed.

**Comment by rgrinberg** [comment]
@snowleopard is this still relevant?

**Comment by snowleopard** [comment]
> @snowleopard is this still relevant?

The corresponding issue is closed, so I guess we no longer need this PR. Furthermore, the current implementation is stateful, so it won't work with the incremental mode. I suggest we archive it for now.

**Comment by snowleopard** [comment]
> Furthermore, the current implementation is stateful, so it won't work with the incremental mode. I suggest we archive it for now.

Actually, it's a warning, not an error, so it's probably OK-ish (if not desirable) to warn the user just once, not in every restart.

**Comment by snowleopard** [comment]
> This could be a `(remove_target_write_permissions enabled/disabled)` stanza in the dune configuration file.

I think I prefer this approach.

**Pull request closed without merging**